File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2817_ALPS식투표/2817.py
X = int(input())
N = int(input())
staff = [tuple(input().split()) for _ in range(N)]
chip_cnt = {chr(i): 0 for i in range(65, 91)}
check = {chr(i): 0 for i in range(65, 91)}
v = []

for i in range(N):
    s, c = staff[i]
    c = int(c)
    if c * 20 >= X:
        check[s] = 1
        for j in range(1, 15):
            v.append((c / j, s))

v.sort(reverse=True)

for i in range(min(len(v), 14)):
    chip_cnt[v[i][1]] += 1

for i in range(65, 91):
    c = chr(i)
    if check[c]:
        print(c, chip_cnt[c])

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/2817_ALPS식투표/2817.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

int X, N;
int chip_cnt[256];
int check[256];
pair<char, int> staff[10];
vector<pair<double, char>> v;

int main()
{
	scanf(" %d", &X);
	scanf(" %d", &N);
	for (int i = 0; i < N; i++)
		scanf(" %c %d", &staff[i].first, &staff[i].second);

	for (int i = 0; i < N; i++)
	{
		if (staff[i].second * 20 >= X)
		{
			check[staff[i].first] = 1;
			for (int j = 1; j <= 14; j++)
				v.push_back({staff[i].second / (double)j, staff[i].first});
		}
	}

	sort(v.begin(), v.end());

	int size = v.size();
	for (int i = 0; i < size && i < 14; i++)
		chip_cnt[v[size - 1 - i].second]++;

	for (int i = 'A'; i <= 'Z'; i++)
		if (check[i])
			printf("%c %d\n", (char)i, chip_cnt[i]);

	return 0;
}